contracts/multisig/src/key.rs

```diff
@@ -13,8 +14,43 @@ pub enum KeyType {
 
 #[cw_serde]
 #[derive(PartialOrd, Ord, Eq)]
-pub enum Signature {
-    Ecdsa(HexBinary),
+pub enum Signature<T>
+where
+    T: Recoverability,
+{
+    Ecdsa(HexBinary, PhantomData<T>),
+}
+
+pub trait Recoverability {
+    fn len() -> usize;
+}
+
+#[cw_serde]
+#[derive(Ord, PartialOrd, Eq)]
+pub struct Recoverable;
+
+impl Recoverable {
+    const LEN: usize = 65;
+}
+
+#[cw_serde]
+#[derive(Ord, PartialOrd, Eq)]
+pub struct NonRecoverable;
+
+impl NonRecoverable {
+    const LEN: usize = 64;
```

Comment:
Are these sizes correct for all signature schemes, or just ecdsa?

Reply:
it might just be ecdsa. I probably have to move the generic element into the variant instead